rocAL/rocAL/include/api/rocal_api_data_loaders.h

```diff
@@ -155,26 +155,26 @@ extern "C"  RocalImage  ROCAL_API_CALL rocalJpegCOCOFileSource(RocalContext cont
 /// \param is_output Determines if the user wants the loaded images to be part of the output or not.
+/// \param area_factor Determines how much area to be cropped. Ranges from from 0.08 - 1.
+/// \param aspect_ratio Determines the aspect ration of crop. Ranges from 0.75 to 1.33.
+/// \param num_attempts Maximum number of attempts to generate crop. Default 10
 /// \param decode_size_policy
 /// \param max_width The maximum width of the decoded images, larger or smaller will be resized to closest
 /// \param max_height The maximum height of the decoded images, larger or smaller will be resized to closest
-/// \param area_factor Determines how much area to be cropped. Ranges from from 0.08 - 1.
-/// \param aspect_ratio Determines the aspect ration of crop. Ranges from 0.75 to 1.33.
-/// \param y_drift_factor - Determines from top left corder to height (crop_height), where to start cropping other wise try for a central crop or take image dims. Ranges from 0 to 1.
-/// \param x_drift_factor - Determines from top left corder to width (crop_width), where to start cropping other wise try for a central crop or take image dims. Ranges from 0 to 1.
 /// \return Reference to the output image
 extern "C"  RocalImage  ROCAL_API_CALL rocalJpegCOCOFileSourcePartial(RocalContext p_context,
                                                             const char* source_path,
                                                             const char* json_path,
                                                             RocalImageColor rocal_color_format,
                                                             unsigned internal_shard_count,
                                                             bool is_output,
+                                                            std::vector<float>& area_factor,
+                                                            std::vector<float>& aspect_ratio,
+                                                            unsigned num_attempts,
                                                             bool shuffle = false,
                                                             bool loop = false,
-                                                            RocalImageSizeEvaluationPolicy decode_size_policy = ROCAL_USE_MAX_SIZE,
-                                                            unsigned max_width = 0, unsigned max_height = 0,
-                                                            RocalFloatParam area_factor = NULL, RocalFloatParam aspect_ratio = NULL,
-                                                            RocalFloatParam y_drift_factor = NULL, RocalFloatParam x_drift_factor = NULL);
+                                                            RocalImageSizeEvaluationPolicy decode_size_policy = ROCAL_USE_USER_GIVEN_SIZE_RESTRICTED,
```

Comment:
Should default decode_size_policy be ROCAL_USE_USER_GIVEN_SIZE_RESTRICTED or ROCAL_USE_MAX_SIZE_RESTRICTED?

Reply:
USER_GIVEN_SIZE_RESTRICTED should be the default policy